### PR TITLE
 [CALCITE-5737] Support jdk21 and jdk22

### PR DESCRIPTION
**Pull request by snuyanzin**
Adds jdk21 and 22 to ci
The PR also adds deprecations to forbiddenAPI config for `URL` constructor which became deprecated since jdk20

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -792,6 +808,9 @@ allprojects {
                 inputs.property("java.vm.version", System.getProperty("java.vm.version"))
                 options.encoding = "UTF-8"
                 options.compilerArgs.add("-Xlint:deprecation")
+                // JDK 1.8 is deprecated https://bugs.openjdk.org/browse/JDK-8173605
+                // and now it requires -Xlint:-options to mute warnings about its deprecation
+                options.compilerArgs.add("-Xlint:-options")
```
Comment (snuyanzin) [inline build.gradle.kts:813]: Approach similar to one done within openJDK https://github.com/openjdk/jdk/pull/10135

**Review by caicancai — COMMENTED** [review]
I'm sure you've tested it all locally

**Comment by snuyanzin** [comment]
>I'm sure you've tested it all locally

yep, exactly, however the main test is ci since locally I have only one OS and on ci there are different

**Comment by snuyanzin** [comment]
I rebased the PR since gradle8 support was merged

**Comment by snuyanzin** [comment]
I will merge tomorrow if there is no objections

**Pull request merged**